### PR TITLE
update hpa apiVersion to autoscaling/v2beta2

### PR DESCRIPTION
**Pull request by tariq1890**
This updates the apiVersion of the HPAs to be used within Istio.

### CODE REVIEW

File: manifests/charts/istio-telemetry/mixer-telemetry/templates/autoscale.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {{ $telemetry := index .Values "mixer" "telemetry" }}
 {{- if and $telemetry.autoscaleEnabled $telemetry.autoscaleMin $telemetry.autoscaleMax }}
-apiVersion: autoscaling/v2beta1
+apiVersion: autoscaling/v2beta2
```
Comment (howardjohn) [inline manifests/charts/istio-telemetry/mixer-telemetry/templates/autoscale.yaml:3]: why not autoscaling/v1?

Comment (tariq1890) [reply manifests/charts/istio-telemetry/mixer-telemetry/templates/autoscale.yaml:3]: I believe this is the reason:

>The beta version, which includes support for scaling on memory and custom metrics, can be found in autoscaling/v2beta2

As seen [here](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/).

Comment (tariq1890) [reply manifests/charts/istio-telemetry/mixer-telemetry/templates/autoscale.yaml:3]: Also, I assumed `v2beta2` would be more of an upgrade from `v2beta1` as opposed to `v1`

Comment (howardjohn) [reply manifests/charts/istio-telemetry/mixer-telemetry/templates/autoscale.yaml:3]: oh of course. I was thinking v1 > v2beta2

Comment (tariq1890) [reply manifests/charts/istio-telemetry/mixer-telemetry/templates/autoscale.yaml:3]: Cool! I'll address the e2e failures then :)

**Review by howardjohn — COMMENTED** [review]
LGTM in general. Is the api the same such that we just need to change the version?

**Comment by tariq1890** [comment]
Just the schema change which you will see shortly after I push the fixes.

**Comment by tariq1890** [comment]
The e2e failures throw this error:

```
 Ingress gateways encountered an error: failed to create "HorizontalPodAutoscaler/istio-system/istio-ingressgateway": HorizontalPodAutoscaler.autoscaling "istio-ingressgateway" is invalid: [spec.metrics[0].resource.target.type: Required value: must specify a metric target type, spec.metrics[0].resource.target.type: Invalid value: "": must be either Utilization, Value, or AverageValue, spec.metrics[0].resource.target.averageUtilization: Required value: must set either a target raw value or a target utilization]
```

But I am not able to understand where I may have missed the value. Everywhere I see the `target.type` has been provided.

**Comment by howardjohn** [comment]
Oh right I forgot about this problem.. https://github.com/istio/istio/blob/5ea94a827beb59213af6a999c2178e7e3aa4b0c1/manifests/profiles/default.yaml#L61. The operator API has a HPA section which has v2beta1 schema

**Comment by howardjohn** [comment]
Fixes https://github.com/istio/istio/issues/19423

**Pull request closed without merging**